### PR TITLE
Refactor: [M3-6280] MUI Migration: Profile

### PR DESCRIPTION
**Pull request by abailly-akamai**
## Description 📝
MUI migration for the whole Profile section

The PR seems large but only features MUI migrations & code cleanup (exports, ordering, prop names). Managed to do it all with styled components and occasional `sx` props, moving us away from makeStyles()

## How to test 🧪
1. navigate /profile
2. go through all flows, compare with production styles
3. confirm no regression was introduced as part of this PR

### CODE REVIEW

File: packages/manager/src/components/GravatarByEmail.tsx [hunk 1]
```diff
@@ -1,36 +1,39 @@
...
-});
+interface Props {
+  height?: number;
+  width?: number;
+}
 
 interface Props {
-  email: string;
   className?: string;
+  email: string;
```
Comment (mjac0bs) [inline packages/manager/src/components/GravatarByEmail.tsx:15]: What's the reason behind declaring two Props here?

**Review by mjac0bs — APPROVED** [review]
Nice, this looks consistent with prod, from what I can tell. I called out one minor difference I noticed.


There are a few things that are issues both in prod and this branch. I'll list them here as observations since they'd probably really be beyond the scope of a migration ticket. We could also separately ticket these - at least the last two we should fix at some point.

Weird animation on the Notifications toggle on the My Settings page that is inconsistent with the TTC toggle on the same page:

https://github.com/linode/manager/assets/114685994/913481bc-4320-4d6f-b31b-a78b44251c64


Close dialogs (e.g. on OAuth Apps tab) extend beyond the viewport on small screens:
image

Weird (API filtering?) issue with TPAs table not being filterable by label, and then results in a "could not apply filter" error state for the table:

image

File: packages/manager/src/components/GravatarByEmail.tsx (hunk 1 above)
Comment (abailly-akamai) [reply packages/manager/src/components/GravatarByEmail.tsx]: no reason, fixed!

File: packages/manager/src/features/Profile/AuthenticationSettings/SMSMessaging.tsx [hunk 2]
```diff
@@ -67,44 +44,36 @@ export const SMSMessaging = () => {
...
     <>
-      <Notice
-        spacingTop={12}
+      <StyledNotice
         spacingBottom={16}
         spacingLeft={1}
+        spacingTop={12}
         success={hasVerifiedPhoneNumber}
         warning={!hasVerifiedPhoneNumber}
-        className={hasVerifiedPhoneNumber ? classes.notice : undefined}
```
[the remark this answers is not in the document]
Comment (abailly-akamai) [reply packages/manager/src/features/Profile/AuthenticationSettings/SMSMessaging.tsx:76]: good catch, fixed!

**Review by bnussman-akamai — APPROVED** [review]
Everything looks solid, just found one thing!

This button does not look flush with the table

Screenshot 2023-06-08 at 10 59 30 AM

**Comment by abailly-akamai** [comment]
@bnussman fixed!

**Pull request merged**